holidays/countries/azerbaijan.py

```diff
@@ -26,54 +27,52 @@ class Azerbaijan(HolidayBase):
     country = "AZ"
 
     def _populate(self, year: int) -> None:
-        def _add_observed(hol_date: date, hol_name: str) -> None:
-            next_workday = hol_date + td(days=+1)
-            while next_workday.year == year and (
-                self._is_weekend(next_workday) or self.get(next_workday)
-            ):
-                next_workday += td(days=+1)
-            _add_holiday(next_workday, f"{hol_name} (Observed)")
-
-        def _add_holiday(hol_date: date, hol_name: str) -> None:
-            if hol_date.year == year:
-                self[hol_date] = hol_name
+        def add_to_list(hol_date: date, hol_name: str) -> None:
+            if hol_date in hol_list:
+                hol_list[hol_date].append(hol_name)
+            else:
+                hol_list[hol_date] = [hol_name]
 
         if year <= 1989:
             return None
 
         super()._populate(year)
 
+        hol_list: Dict[date, List[str]] = {}
```

Comment:
A bit misleading naming (list vs dict). I prefer using something like `holiday_date_names_mapping`.

Reply:
Done. Thank you!